<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1119</width>
    <height>809</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>-1</x>
      <y>0</y>
      <width>1121</width>
      <height>811</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="mainMenuPage">
     <property name="styleSheet">
      <string notr="true">border-image: url(&quot;resource/image/background.jpg&quot;) 0 0 0 0 stretch stretch;</string>
     </property>
     <widget class="QLabel" name="mainTitle">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>160</y>
        <width>471</width>
        <height>71</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Courier New</family>
        <pointsize>30</pointsize>
        <weight>50</weight>
        <italic>false</italic>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: rgba(255, 255, 255, 0.8);
border-image: none;
font: 30pt &quot;Courier New&quot;;</string>
      </property>
      <property name="text">
       <string>Halma Game</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QGroupBox" name="mainMenuGroup">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>300</y>
        <width>291</width>
        <height>221</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">border-image: none;
border: none;
font: 20pt &quot;Courier New&quot;;</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="exitBtn">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(255, 0, 0);
color: #ffffff;
border: 3px solid #000;
border-radius: 30px;</string>
       </property>
       <property name="text">
        <string>Exit</string>
       </property>
      </widget>
      <widget class="QPushButton" name="playGameBtn">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(170, 255, 0);
border-radius: 30px;
border: 3px solid #000;
</string>
       </property>
       <property name="text">
        <string>Play Game</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="selectBoardSizeMenuPage">
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 255, 127);</string>
     </property>
     <widget class="QLabel" name="selectSizeTitle">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>100</y>
        <width>521</width>
        <height>91</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Courier New</family>
        <pointsize>30</pointsize>
        <weight>50</weight>
        <italic>false</italic>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: none;
border-image: none;
font: 30pt &quot;Courier New&quot;;</string>
      </property>
      <property name="text">
       <string>Select Board Size</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QGroupBox" name="selectSizeGroup">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>260</y>
        <width>291</width>
        <height>321</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background: none;
border: none;
font: 24pt &quot;Courier New&quot;;</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="eight">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(0, 255, 127);
border: 3px solid #000;
border-radius: 30px;</string>
       </property>
       <property name="text">
        <string>8 x 8</string>
       </property>
      </widget>
      <widget class="QPushButton" name="ten">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(0, 255, 255);
border: 3px solid #000;
border-radius: 30px;</string>
       </property>
       <property name="text">
        <string>10 x 10</string>
       </property>
      </widget>
      <widget class="QPushButton" name="sixteen">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>220</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(255, 85, 0);
border: 3px solid #000;
border-radius: 30px;
color: #ffffff;</string>
       </property>
       <property name="text">
        <string>16 x 16</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="selectGameModeMenuPage">
     <property name="styleSheet">
      <string notr="true">background-color: rgb(170, 170, 255);</string>
     </property>
     <widget class="QGroupBox" name="selectModeGroup">
      <property name="geometry">
       <rect>
        <x>338</x>
        <y>260</y>
        <width>441</width>
        <height>321</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background: none;
border: none;</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="humanVsMinimax">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>401</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgb(170, 255, 127), stop:1 rgb(255, 170, 255));
border: 3px solid #000;
border-radius: 30px;
font: 10pt &quot;Courier New&quot;;</string>
       </property>
       <property name="text">
        <string>Human vs Bot Minimax</string>
       </property>
      </widget>
      <widget class="QPushButton" name="humanVsLocalSearch">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>401</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgb(170, 255, 127), stop:1 rgb(170, 255, 255));
border: 3px solid #000;
border-radius: 30px;
font: 10pt &quot;Courier New&quot;;</string>
       </property>
       <property name="text">
        <string>Human vs Bot Minimax with Local Search</string>
       </property>
      </widget>
      <widget class="QPushButton" name="minimaxVsLocalSearch">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>220</y>
         <width>401</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, stop:0 rgb(255, 170, 255), stop:1 rgb(170, 255, 255));
border: 3px solid #000;
border-radius: 30px;
font: 10pt &quot;Courier New&quot;;</string>
       </property>
       <property name="text">
        <string>Bot Minimax vs Bot Minimax Local Search</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="selectGameModeTitle">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>120</y>
        <width>501</width>
        <height>71</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Courier New</family>
        <pointsize>30</pointsize>
        <weight>50</weight>
        <italic>false</italic>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: none;
border-image: none;
font: 30pt &quot;Courier New&quot;;</string>
      </property>
      <property name="text">
       <string>Select Game Mode</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="selectSideMenuPage">
     <property name="styleSheet">
      <string notr="true">background-color: rgb(170, 255, 255);</string>
     </property>
     <widget class="QLabel" name="selectSideTitle">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>120</y>
        <width>501</width>
        <height>71</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Courier New</family>
        <pointsize>30</pointsize>
        <weight>50</weight>
        <italic>false</italic>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: none;
border-image: none;
font: 30pt &quot;Courier New&quot;;</string>
      </property>
      <property name="text">
       <string>Select Your Side</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QGroupBox" name="selectSideGroup">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>260</y>
        <width>291</width>
        <height>321</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 12pt &quot;Courier New&quot;;
background: none;
border: none;</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QPushButton" name="pGreenBtn">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(170, 255, 0);
border: 3px solid #000;
border-radius: 30px;
</string>
       </property>
       <property name="text">
        <string>Player Green</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pRedBtn">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(255, 85, 127);
border: 3px solid #000;
border-radius: 30px;
</string>
       </property>
       <property name="text">
        <string>Player Red</string>
       </property>
      </widget>
      <widget class="QPushButton" name="mainMenuNavBtn">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>220</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(255, 85, 0);
border: 3px solid #000;
border-radius: 30px;
color: #ffffff;</string>
       </property>
       <property name="text">
        <string>Back to Main Menu</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="maxTimePage">
     <property name="styleSheet">
      <string notr="true">background-color: rgb(119, 255, 153);</string>
     </property>
     <widget class="QGroupBox" name="maxTimeGroup">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>260</y>
        <width>291</width>
        <height>321</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 12pt &quot;Courier New&quot;;
background: none;
border: none;</string>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QSpinBox" name="maxTime">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>140</y>
         <width>100</width>
         <height>50</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Courier New</family>
         <pointsize>12</pointsize>
         <weight>50</weight>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Only insert number</string>
       </property>
       <property name="toolTipDuration">
        <number>-1</number>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
      <widget class="QPushButton" name="startGameButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>220</y>
         <width>251</width>
         <height>81</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">background: rgb(170, 255, 0);
border-radius: 30px;
border: 3px solid #000;
</string>
       </property>
       <property name="text">
        <string>Start Game</string>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="maxTimeTitle">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>120</y>
        <width>501</width>
        <height>71</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Courier New</family>
        <pointsize>30</pointsize>
        <weight>50</weight>
        <italic>false</italic>
        <bold>false</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background: none;
border-image: none;
font: 30pt &quot;Courier New&quot;;</string>
      </property>
      <property name="text">
       <string>Input Max Time</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="inGamePage">
     <property name="styleSheet">
      <string notr="true">background-color: rgb(85, 170, 255);</string>
     </property>
     <widget class="QLabel" name="fieldsLabel">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>70</y>
        <width>701</width>
        <height>701</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget">
      <property name="geometry">
       <rect>
        <x>49</x>
        <y>79</y>
        <width>681</width>
        <height>681</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="fields">
       <property name="spacing">
        <number>0</number>
       </property>
      </layout>
     </widget>
     <widget class="QLabel" name="curPlayer">
      <property name="geometry">
       <rect>
        <x>800</x>
        <y>90</y>
        <width>231</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Segoe UI Semibold</family>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="quitGameBtn">
      <property name="geometry">
       <rect>
        <x>860</x>
        <y>150</y>
        <width>141</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Quit Game</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
